core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -1164,7 +1176,8 @@ public RelBuilder filter(Iterable<? extends RexNode> predicates) {
 
     if (!simplifiedPredicates.isAlwaysTrue()) {
       final Frame frame = stack.pop();
-      final RelNode filter = filterFactory.createFilter(frame.rel, simplifiedPredicates);
+      final RelNode filter = filterFactory.createFilter(frame.rel,
+          simplifiedPredicates, Lists.newArrayList(correlVariables));
```

Comment:
Usage of `Lists#newArrayList` is deprecated. Use `new ArrayList<>()` instead.

It feels like `FilterFactory` should actually accept `Iterable<? extends RexNode>` predicates too, as well as constructors, to avoid unnecessary copies.